core/src/main/java/com/yahoo/imapnio/async/response/ImapResponseMapper.java

```diff
@@ -377,6 +381,44 @@ private Status parseToStatus(@Nonnull final IMAPResponse[] r) throws ParsingExce
             return status;
         }
 
+        /**
+         * Parses the LIST-STATUS command responses to a ListStatusResult object. See RFC 5819 for details. For each selectable mailbox matching the
+         * list pattern and selection options, the server MUST return an untagged LIST response followed by an untagged STATUS response containing the
+         * information requested in the STATUS return option. If an attempted STATUS for a listed mailbox fails because the mailbox can't be selected
+         * , the STATUS response MUST NOT be returned and the LIST response MUST include the \NoSelect attribute.
+         *
+         * @param r the list of responses from Status command, the input responses array should contain the tagged/final one
+         * @return Status object constructed based on the r array
+         * @throws ParsingException when encountering parsing exception
+         * @throws ImapAsyncClientException when input value is not valid
+         */
+        @Nonnull
+        private ListStatusResult parseToListStatusResult(@Nonnull final IMAPResponse[] r) throws ParsingException, ImapAsyncClientException {
+            if (r.length < 1) {
+                throw new ImapAsyncClientException(FailureType.INVALID_INPUT);
+            }
+            final Response taggedResponse = r[r.length - 1];
+            if (!taggedResponse.isOK()) {
+                throw new ImapAsyncClientException(FailureType.INVALID_INPUT);
+            }
+            // command successful reaching here
+            final List<ListInfo> v = new ArrayList<ListInfo>();
+            final Map<String, Status> m = new HashMap<>();
+
+            for (int i = 0; i < r.length; i++) {
+                final IMAPResponse ir = r[i];
+
+                if (ir.keyEquals("LIST")) {
+                    v.add(new ListInfo(ir));
```

Comment:
One more issue I realized by looking at sun code is that it does not handle \NonExistent. So ListInfo.canOpen will be set to true but you won't have corresponding Status class. You might have to extend ListInfo and look for the \NonExistent attribute.

Reply:
Done. Created ExtensionListInfo to provide easy access whether those attributes present.